### PR TITLE
New testcases to get more coverage on hera

### PR DESCRIPTION
**Pull request by hugo-dc**
New test cases to get more coverage on hera:

- callValueInStaticMode
- ~createCodeWithRevert~ - create is returning `0x0100000000000000000000000000000000000000000000000000000000000000` instead of `0x01`, check new PR https://github.com/ewasm/tests/pull/68
- getBlockHashOOB
- logZeroLengthMemory
- malformedBytecodeInvalidPreamble
- malformedBytecodeInvalidWasm
- selfDestructExistingAccount
- selfDestructNewAccount

### CODE REVIEW

File: src/GeneralStateTestsFiller/stEWASMTests/malformedBytecodeInvalidWasmFiller.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  pre:
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '100000000000'
+      code: ''
+      nonce: '0'
+      storage: {}
+    b94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '100000000000'
+      # this bytecode is the same as in getAddress
+      # changed this byte from 10 to 11
```
Comment (jakelang) [inline src/GeneralStateTestsFiller/stEWASMTests/malformedBytecodeInvalidWasmFiller.yml]: Perhaps change more than one byte, just to increase the chance Binaryen will throw parseException?

File: src/GeneralStateTestsFiller/stEWASMTests/selfDestructExistingAccountFiller.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      code: ''
+      nonce: '0'
+      storage: {}
+    b94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '100000000000'
+      code: |
+        (module
+          (import "ethereum" "selfDestruct" (func $selfDestruct (param i32)))
+          (memory 1)
+          (data (i32.const 0) "\80\75\8d\9f\09\8c\e0\89\9d\c0\70\1f\9d\ad\fc\f4\ad\45\8d\c8")
```
Comment (jakelang) [inline src/GeneralStateTestsFiller/stEWASMTests/selfDestructExistingAccountFiller.yml]: I think the beneficiary should be the `a94...0b` address as it is executing the transaction and the hackmd says it should self-destruct back into the original sender.

File: src/GeneralStateTestsFiller/stEWASMTests/selfDestructNewAccountFiller.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# selfDestruct new account
+---
+selfDestructNewAccount:
```
Comment (jakelang) [inline src/GeneralStateTestsFiller/stEWASMTests/selfDestructNewAccountFiller.yml]: I think this was already done, but under a different name. The `selfDestructNewAccount` case seems to be covered by the basic interface test?

File: src/GeneralStateTestsFiller/stEWASMTests/selfDestructNewAccountFiller.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    currentTimestamp: '1000'
+    previousHash: 5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6
+  pre:
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '100000000000'
+      code: ''
+      nonce: '0'
+      storage: {}
+    b94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '100000000000'
```
Comment (jakelang) [inline src/GeneralStateTestsFiller/stEWASMTests/selfDestructNewAccountFiller.yml]: Should the balance be 0 here?

**Comment by hugo-dc** [comment]
@jakelang thanks for reviewing, I've changed more bytecodes in the `malformedBytecodeInvalidWasm` test case, also removed `selfDestructNewAccount` testcase because as you said another test case already exists for this I just took the previous test case and changed it according to the spec. Also made changes to `selfDestructNewAccount" to follow the specification.

**Review by jakelang — APPROVED** [review]
Looks good!

**Pull request merged**